command/issues/list/list.go

```diff
@@ -145,11 +131,40 @@ func (opts *IssuesListOptions) getIssuesData(ctx context.Context) (err error) {
 
 	// Fetch issues for a certain FileArg (filepath) passed by the user
 	// Example: `deepsource issues list api/hello.py`
-	if opts.FileArg != "" {
-		opts.issuesData, err = deepsource.GetIssuesForFile(ctx, opts.SelectedRemote.Owner, opts.SelectedRemote.RepoName, opts.SelectedRemote.VCSProvider, opts.FileArg, opts.LimitArg)
+	if len(opts.FileArg) != 0 {
+		var fetchedIssues []issues.Issue
+		var filteredIssues []issues.Issue
+
+		opts.issuesData, err = deepsource.GetIssues(ctx, opts.SelectedRemote.Owner, opts.SelectedRemote.RepoName, opts.SelectedRemote.VCSProvider, opts.LimitArg)
 		if err != nil {
 			return err
 		}
+
+		for _, arg := range opts.FileArg {
+			// use regex to segregate directory and file names
+			var pathRegexp = regexp.MustCompile(`^(.*/)?(?:$|(.+?)(?:(\.[^.]*$)|$))`)
+			matched := pathRegexp.FindStringSubmatch(arg)
+
+			// if the argument contains a directory, fetch issues for the directory
+			if matched[1] != "" {
+				filteredIssues, err = filterIssuesByPath(arg, opts.issuesData)
+				if err != nil {
+					return err
+				}
+
+				fetchedIssues = append(fetchedIssues, filteredIssues...)
+				continue
+			}
+
+			filteredIssues, err = deepsource.GetIssuesForFile(ctx, opts.SelectedRemote.Owner, opts.SelectedRemote.RepoName, opts.SelectedRemote.VCSProvider, arg, opts.LimitArg)
```

Comment:
Do we still need this when we are already fetching issues for the whole project? Is it because of the 100 limit?

Reply:
We need this when the user uses only file names as arguments. Since the above condition only works for directories, we need this to fetch file-related issues.

Reply:
Yep, but since we are already fetching all the issues in line 138, thus we have all the data. We can filter for files as well no since `filterIssuesByPath` works for files as well. I mean we don't need the regex condition now, isn't it?

Reply:
Makes sense. Updating the PR.